warden/x/warden/keeper/msg_server_new_sign_request.go

```diff
@@ -66,3 +62,26 @@ func (k msgServer) NewSignRequest(ctx context.Context, msg *types.MsgNewSignRequ
 
 	return &types.MsgNewSignRequestResponse{Id: id}, nil
 }
+
+func chargeKeychainFee(k *msgServer, ctx context.Context, requestedFee string, keychainAccAddress sdk.AccAddress, keychainFees sdk.Coins, creator string) error {
```

Comment:
```suggestion
func (k Keeper) deductKeychainFees(ctx context.Context, requestedFee string, keychainAccAddress sdk.AccAddress, keychainFees sdk.Coins, creator string) error {
```

suggesting to make this a method of Keeper, instead of just a function, taken without pointer. And to rename it to use the verb deduct which is more widely used across the cosmos sdk codebase (like in the DeductFeeDecorator)

could be even worth moving this file into a new `x/warden/keeper/keychain.go` file

Reply:
✅